plotly_express/_core.py

```diff
@@ -74,9 +62,17 @@ def get_trendline_results(fig):
 
 Mapping = namedtuple(
     "Mapping",
-    ["show_in_trace_name", "grouper", "val_map", "sequence", "updater", "variable"],
+    [
+        "show_in_trace_name",
+        "grouper",
+        "val_map",
+        "sequence",
+        "updater",
+        "variable",
+        "facet",
+    ],
 )
-TraceSpec = namedtuple("TraceSpec", ["constructor", "attrs", "trace_patch"])
+TraceSpec = namedtuple("TraceSpec", ["constructor", "attrs", "trace_patch", "marginal"])
```

Comment:
This is a good idea, and maybe should be generalized later to model the "role" of the trace in the figure... Is it the main event, or a marginal, or a trendline, or some other future thing like the raw scatter that someone might want to add to a density heatmap?